### PR TITLE
Accept file and path names to be pathlib objects

### PR DESCRIPTION
**Pull request by JuliaSprenger**
All IOs implicitly assume that the file and directory names provided are strings. Therefore many IOs fail, when a path specified as a [pathlib](https://docs.python.org/3/library/pathlib.html) object is provided instead. This PR introduces the automatic conversion of filenames and directory names, for all IOs and adds a common IO test.

### CODE REVIEW

File: neo/test/iotest/common_io_test.py [hunk 1]
```diff
@@ -192,65 +194,6 @@ def generic_io_object(self, filename=None, return_path=False, clean=False):
                                         return_path=return_path,
                                         clean=clean)
 
-    def create_file_reader(self, filename=None, return_path=False,
```
Comment (samuelgarcia) [inline neo/test/iotest/common_io_test.py:195]: I don't remember why we add this before.
it is not used anymore ?

**Comment by samuelgarcia** [comment]
Ok for me.

File: neo/test/iotest/common_io_test.py (hunk 1 above)
Comment (JuliaSprenger) [reply neo/test/iotest/common_io_test.py:195]: It is not used within Neo and is also not working, e.g. when trying to create a generic reader as the default filename does not exist.

**Comment by JuliaSprenger** [comment]
I am not sure why circleCI is not receiving output for some time. I think we should investigate this phenomenon in general, since this has been an issue also in other PRs... Maybe it's a matter of adjusting the CI configuration to our current requirements?

**Comment by JuliaSprenger** [comment]
This is rebased and tests are fixed. @samuelgarcia ready for final review.

**Comment by samuelgarcia** [comment]
OK for me.
I merge.

**Pull request merged**